cairo_programs/cairo_0/bitwise_builtin_test.cairo

```diff
@@ -4,17 +4,11 @@
 from starkware.cairo.common.bitwise import bitwise_and, bitwise_xor, bitwise_or, bitwise_operations
 from starkware.cairo.common.cairo_builtins import BitwiseBuiltin
 
-func main{bitwise_ptr: BitwiseBuiltin*}() {
-    let (and_a) = bitwise_and(12, 10);  // Binary (1100, 1010).
-    assert and_a = 8;  // Binary 1000.
-    let (xor_a) = bitwise_xor(12, 10);
-    assert xor_a = 6;
-    let (or_a) = bitwise_or(12, 10);
-    assert or_a = 14;
+func main{bitwise_ptr: BitwiseBuiltin*}(x: felt, y: felt) {
+
+    assert bitwise_ptr.x = x;
+    assert bitwise_ptr.y = y;
+    tempvar and = bitwise_ptr.and
```

Comment:
Don't use `tempvar`, it creates a temporary variable, see [this](https://docs.cairo-lang.org/how_cairo_works/consts.html) for more details.

Either use references (`let`) and then `assert` or directly use `assert`

You cannot access the result of `x & y` with `bitwise_ptr.and`, you can find the memory structure of the bitwise builtin [here](https://github.com/starkware-libs/cairo-lang/blob/efa9648f57568aad8f8a13fbf027d2de7c63c2c0/src/starkware/cairo/common/cairo_builtins.cairo#L18-L25)

Reply:
Ohh now it's clear for me! Thank you very much for the resources!

Reply:
This should be good now ✅ 
But I have a question I think we are doing pretty much the same in `bitwise_test.cairo`, in the issue you mentioned cleaning up the cairo files, so, should I delete the `bitwise_test.cairo` file?

Reply:
Done! ✅ 
 I do have a question when I tried to run `bun test` it failed for `cairoRunner.test.ts` with `No such file or directory` when trying to read `'cairo_programs/cairo_0/bitwise_builtin.json'`, is this expected? is `.json` the proper extension?

Reply:
You need to compile the cairo programs before executing them in the VM

It can be done with `make compile` (needs poetry & python ^3.11) or directly using the [cairo zero compiler](https://docs.cairo-lang.org/quickstart.html)